aoservicelisting.civix.php

```diff
@@ -76,6 +76,102 @@ public static function findClass($suffix) {
     return self::CLASS_PREFIX . '_' . str_replace('\\', '_', $suffix);
   }
 
+  public static function verifyListing($duration) {
+    // Get a list of service listings who have been approved with the latest approval activity.
+    // Duration is specified in months, so multiply by 4.345
+    $weeks = (integer) ($duration * 4.345) - 1;
+    if (empty($weeks)) {
+      $weeks = 25;
+    }
+    $result = [];
+    $sql = "SELECT ac.contact_id, DATE(a.activity_date_time) as activity_date
+      FROM civicrm_activity_contact ac
+      INNER JOIN civicrm_activity a ON a.id = ac.activity_id
+      INNER JOIN civicrm_value_service_listi_71 s ON s.entity_id = ac.contact_id
+      WHERE ac.record_type_id = 3
+      AND a.is_current_revision = 1
+      AND s.service_provider_status_872 = 'Approved'
+      AND a.activity_type_id = 142
+      AND DATE_ADD(DATE(a.activity_date_time), INTERVAL $weeks WEEK) = DATE(NOW())
+      AND a.subject = \"Application status changed to Approved\"
+      AND a.id IN
+      (SELECT MAX(a.id)
```

Comment:
@Edzelopez on this one maybe we need to do a.id, max(a.activity_date_time) here in case activity.ids aren't in chronological order